### DISCUSSION TITLE
Completion: Full description length

### DISCUSSION THREAD

**Discussion by dezza** [akinomyoga/ble.sh#532] [Q&A] [answered]
Hi, is it possible to get full description of arguments?

For example with `ls --<TAB>`

Got:
```
--dereference                             : [-L] when showing file information for a symbolic link, show information for the file the link references rath…
```

Expected:
```
--dereference                             : [-L] when showing file information for a symbolic link, show information for the file the link references rather than for the link itself
```

The description is limited and cut off by `...`

I tried the settings, `complete_menu_style` none of them did it, I have disabled `menu_desc_multicolumn_width=`, tried searching a bit in the code but its quite overwhelming and don't know here to look..

I understand it might be a general limit, just to avoid extremely lengthy descriptions.. Maybe it could be a setting to configure to allow multi-line descriptions and then set a line-limit or total characters.

**Comment by akinomyoga** [comment]
Thanks. Hmm, with `bleopt menu_desc_multicolumn_width=`, the truncation should only happen on the right edge of the terminal by design. If it is truncated in the middle of the line in the terminal, it would be a bug in the calculation of the terminal width.

- **Q1**: Can you confirm that the truncation happens in the middle of the line in your environment?
- **Q2**: What are the results of the following commands?

```bash
$ echo "$LINES $COLUMNS"
$ stty size
```

**Reply by dezza** [reply]
Yes it does only happen on the right side, my question is whether you can disable this and make it continue on the next line (showing full description spanning 2+ lines) :)

~~Side-question: also is it possible to disable the "(0: << !1016   ) `'" status message when using C-R substring search?~~
Sry missed: https://github.com/akinomyoga/ble.sh/issues/80 found it: https://github.com/akinomyoga/ble.sh/issues/101#issuecomment-828541531

**Reply by akinomyoga** [reply, marked as answer]
> Yes it does only happen on the right side, my question is whether you can disable this and make it continue on the next line (showing full description spanning 2+ lines) :)

It's not implemented currently, and I need to newly implement it. However, it's not simple to implement. To offer the TUI-like behavior `ble.sh` needs to calculate the layout by emulating/predicting all the text reflow of the terminal. `ble.sh` actually already contains a framework to calculate it, but I anyway need to design new layouting to implement the behavior you describe. I'll put it on my ToDo list, but I don't have time to implement it soon.

> ~Side-question: also is it possible to disable the "(0: << !1016 ) `'" status message when using C-R substring search?~ Sry missed: #80 found it: [#101 (comment)](https://github.com/akinomyoga/ble.sh/issues/101#issuecomment-828541531)

Ah, #80 and #101 are for `nsearch` (non-incremental search), but `C-r` is `isearch` (incremental search). Currently, there is no option to turn off the search progress of `isearch`. You may overwrite a function responsible for showing the progress:

```bash
# blerc

function ble-edit/isearch/.show-status-with-progress.fib { return 0; }
```

**Reply by dezza** [reply]
Great project. Well done, sir! Thanks for the quick response :)
